### PR TITLE
addded basics for angular in topics/techstacks

### PR DESCRIPTION
**Pull request by khushaal-nandwani**

### CODE REVIEW

**Comment by khushaal-nandwani** [comment]
@thakertanya please review

File: Topics/Tech_Stacks/Angular Basics.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+### Communication
+- **Input**: Components can receive data from their parent using `@Input()`. It's like getting a present; the parent component sends it, and the child receives it.
+- **Output**: To communicate back to the parent, a child component uses `@Output()` and EventEmitter. It's like the child sending a letter back home.
+
+Imagine you wanted to dynamically input users into your list component and output the oldest user. You could do something like this:
+
+**app.component.html**
+```
+<app-list [users]="usersFromAppComponent" (oldestUser)="oldestUser($event)"></app-list>
```
Comment (thakertanya) [inline Topics/Tech_Stacks/Angular Basics.md:69]: It might make it clearer if you explain what $event does/is

**Comment by khushaal-nandwani** [comment]
I have added the prerequisites and additional resources. Incorporating best practices and a comprehensive example, I feel, would be too heavy for a basics article. The goal of my article is to simply familiarize a person with the basics of Angular, not to guide them in setting up and running their first Angular tutorial. Therefore, I think it would be best not to include these elements.

**Comment by khushaal-nandwani** [comment]
> * Consider outlining prerequisites for running your examples successfully.
> * Additionally, incorporating best practices in Angular might also be helpful like file structure and naming conventions, if any.
> * I recommend showcasing a comprehensive, end-to-end example that integrates components, services, and modules. This practical demonstration would give readers a clearer insight into how these concepts work together.
> * Lastly, adding links to official angular documentation for additional resources might be helpful.

I have added the prerequisites and additional resources. Incorporating best practices and a comprehensive example, I feel, would be too heavy for a basics article. The goal of my article is to simply familiarize a person with the basics of Angular, not to guide them in setting up and running their first Angular tutorial. Therefore, I think it would be best not to include these elements.

File: Topics/Tech_Stacks/Angular Basics.md (hunk 1 above)
Comment (khushaal-nandwani) [reply Topics/Tech_Stacks/Angular Basics.md:74]: Yes, that'd be helpful. Added the explaination

File: Topics/Tech_Stacks/Angular Basics.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+**list.component.ts** (files are stored as .ts, guess what ts stands for - hint: it's not Top Secret)
+```
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-list',
+  template: `list.component.html`,
+})
+export class ListComponent {
+    users = {
```
Comment (thakertanya) [inline Topics/Tech_Stacks/Angular Basics.md:29]: Just checking, does this need to be a square bracket?

Comment (khushaal-nandwani) [reply Topics/Tech_Stacks/Angular Basics.md:29]: Yes, it should be. Thanks for pointing it out!

**Pull request merged**